--- libmali-0c1c1b8e3a687fd10df38935c4dff9ab90285a09/meson.build.orig	2024-03-20 03:46:59.917531600 +0300
+++ libmali-0c1c1b8e3a687fd10df38935c4dff9ab90285a09/meson.build	2024-03-20 03:48:46.863252077 +0300
@@ -26,7 +26,12 @@
 message('Building for ' + '|'.join([arch, gpu, version, subversion, platform, optimize]))
 
 # Grab libraries with specified configs
-cmd = run_command('scripts/grabber.sh', arch, gpu, version, subversion, platform, optimize)
+if platform == 'wayland'
+  gplatform = 'wayland-gbm'
+else
+  gplatform = platform
+endif
+cmd = run_command('scripts/grabber.sh', arch, gpu, version, subversion, gplatform, optimize)
 libs = cmd.stdout().strip().split('\n')
 
 # Use the first one as default library
